src/pages/map/map.spec.ts

```diff
@@ -0,0 +1,54 @@
+// Mocha global variables (for Windows)
+/// <reference path="../../../node_modules/@types/mocha/index.d.ts" />
+
+import { TestBed } from '@angular/core/testing';
+import { LeafletModule } from '@asymmetrik/ngx-leaflet';
+import { expect } from 'chai';
+import { IonicModule, NavController } from 'ionic-angular';
+import * as L from 'leaflet';
+
+import { translateModuleForRoot } from '../../utils/i18n';
+import { MapPage } from './map';
+
+describe('MapPage', function () {
+  let component;
+  let navControllerMock;
+
+  beforeEach(function () {
+    navControllerMock = {};
+
+    TestBed.configureTestingModule({
+      declarations: [
+        MapPage
+      ],
+      imports: [
+        IonicModule.forRoot(MapPage),
+        translateModuleForRoot,
+        LeafletModule.forRoot()
+      ],
+      providers: [
+        { provide: NavController, useValue: navControllerMock }
+      ]
+    });
+
+  });
+
+  afterEach(function () {
+    component = null;
+  });
+
+  it('should be created', function () {
+    component = TestBed.createComponent(MapPage).componentInstance;
+    expect(component instanceof MapPage).to.equal(true);
+  });
+
+  it('should have a valid mapOptions property', function () {
+    component = TestBed.createComponent(MapPage).componentInstance;
+    expect(component, 'MapPage has no mapOptions property').to.haveOwnProperty('mapOptions');
+    expect(component.mapOptions, 'MapPage.mapOptions has no layers property').to.haveOwnProperty('layers');
+    expect(component.mapOptions.layers, 'MapPage.mapOptions.layers is not an array').to.be.an('array');
+    component.mapOptions.layers.forEach(layer => {
+      expect(layer instanceof L.TileLayer, 'One of MapPage.mapOptions.layers\' item is not an instance of L.TileLayer').to.equal(true);
```

Comment:
You can `expect(someObject).to.be.an.instanceof(someClass)` which produces a more readable default error.

Reply:
And render the test more readable.
Will do.